components/CanvasOne.js

```diff
@@ -0,0 +1,82 @@
+import { useRef, useEffect, useState } from "react";
+import styled from "styled-components";
+
+function lineWidthRandomizer() {
+  const randomNumber = Math.random();
+  if (randomNumber <= 0.4) {
+    return 2;
+  } else if (randomNumber <= 0.8) {
+    return 8;
+  } else {
+    return 4;
+  }
+}
+
+let firstLineWidth = lineWidthRandomizer();
+let secondLineWidth = lineWidthRandomizer();
+let thirdLineWidth = lineWidthRandomizer();
+let fourthLineWidth = lineWidthRandomizer();
+
+export default function CanvasOne() {
+  const canvasRef = useRef(null);
+  const [backgroundColor, setBackgroundColor] = useState(148);
+  useEffect(() => {
+    const canvas = canvasRef.current;
+    const ctx = canvas.getContext("2d");
+    // Beginn drawing statements
+    ctx.fillStyle = `hsl(${backgroundColor}, 60%, 95%)`;
+    ctx.fillRect(0, 0, canvas.width, canvas.height);
+
+    ctx.fillStyle = "hsl(148, 100%, 50%)";
+    ctx.translate(150, 150);
+    ctx.rotate((0 * Math.PI) / 180);
+    ctx.fillRect(-40, -40, 80, 80);
+
+    ctx.lineWidth = firstLineWidth;
+
+    ctx.strokeStyle = `hsl(${backgroundColor}, 60%, 95%)`;
+    ctx.strokeRect(-22.5, -22.5, 45, 45);
+
+    ctx.strokeStyle = "hsl(148, 100%, 50%)";
+    ctx.lineWidth = secondLineWidth;
+    ctx.strokeRect(-60, -60, 120, 120);
+
+    ctx.lineWidth = thirdLineWidth;
+    ctx.strokeRect(-80, -80, 160, 160);
+
+    ctx.lineWidth = fourthLineWidth;
+    ctx.strokeRect(-100, -100, 200, 200);
+
+    // Reset the transformation matrix to the identity matrix
+    ctx.setTransform(1, 0, 0, 1, 0, 0);
+    // End drawing statements
+  }, [backgroundColor]);
+  return (
+    <>
+      <StyledHeadline>Generator 1</StyledHeadline>
+      <StyledCanvas width="300" height="300" ref={canvasRef} />
+      <lable htmlFor="colorSelector">Background color </lable>
+      <input
+        type="range"
+        min="0"
+        max="360"
+        value={backgroundColor}
+        onChange={(event) => setBackgroundColor(event.target.value)}
```

Comment:
when we are right, you are done with the backgroundcolorcheck, with this short onchange function, do you really need line25 to 53 in your code right now?

Reply:
Line 25 - 56 is about creating a background with a color and the graphic elements. The onChange function is about to change the color of the Background after it was created. So you need line 25 - 56 to create the background and the graphic. Both are part of the user story.